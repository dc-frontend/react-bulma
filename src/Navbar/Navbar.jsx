import React from 'react'
import PropTypes from 'prop-types'

/**
 * Brand
 * @param { Object } props
 * @param { String='/' } props.href - Location where the brand should link to
 * @param { String } props.title - Brand name
 */
export function Brand (props) {
  return <div className='navbar-brand'>
    <a href={props.href ? props.href : '/'} className='navbar-item'>{props.title}</a>
  </div>
}
Brand.propTypes = {
  title: PropTypes.string.isRequired,
  href: PropTypes.string
}

/**
 * Dropdown
 * @param { Object } props
 * @param { String } props.title - Dropdown link title
 * @param { Object } props.children - Navbar Items
 */
export function Dropdown (props) {
  return <div className="navbar-item has-dropdown is-hoverable">
    <a className="navbar-link">
      {props.title}
    </a>

    <div className="navbar-dropdown">
      {props.children}
    </div>
  </div>
}
Dropdown.propTypes = {
  children: PropTypes.any,
  title: PropTypes.string.isRequired
}

/**
 * Navbar End
 * Items at the end of the navbar (right side).
 * @param {*} props
 */
export function End (props) {
  return <div className='navbar-end'>
    {props.children}
  </div>
}
End.propTypes = {
  children: PropTypes.any
}

/**
 * Navbar Start
 * Items on the left side of the navbar.
 * @param {*} props
 */
export function Start (props) {
  return <div className='navbar-start'>
    {props.children}
  </div>
}
Start.propTypes = {
  children: PropTypes.any
}

/**
 * Navbar Item
 * @param {*} props
 * @param { String } props.href - Link location.
 * @param { String } props.title - Link title.
 */
export function Item (props) {
  return <a href={props.href} className={`navbar-item ${props.className}`}>{props.title}</a>
}
Item.propTypes = {
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  href: PropTypes.string.isRequired
}

/**
 * Navbar
 * @param {*} props
 */
export function Navbar (props) {
  return <div className={props.dark ? 'navbar is-dark' : 'navbar'}>
    {props.children}
  </div>
}
Navbar.propTypes = {
  children: PropTypes.any,
  dark: PropTypes.bool
}

export default Navbar
