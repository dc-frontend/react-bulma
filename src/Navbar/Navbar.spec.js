// Link.react.test.js
import React from 'react'
import { Navbar } from './Navbar'
import renderer from 'react-test-renderer'

function renderNavbar () {
  const component = renderer.create(
    <Navbar></Navbar>
  )
  return component
}

test('Breadcrumb renders as expected', () => {
  const component = renderNavbar()

  expect(component.toJSON()).toMatchSnapshot()
})
