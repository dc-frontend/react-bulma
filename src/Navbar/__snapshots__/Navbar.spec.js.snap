// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders as expected 1`] = `
<div
  className="navbar"
/>
`;
